nmdc_automation/re_iding/scripts/re_id_tool.py

```diff
@@ -808,5 +913,114 @@ def _get_has_input_from_read_qc(api_client, legacy_id):
     return list(has_input_data_objects)
 
 
+def _update_study_record(study_record: dict, new_study_id: str, db_client: Database[Union[Mapping[str, Any], Any]], no_update: bool) -> dict:
```

Comment:
this function is missing logic to deal with igsn: and emsl: prefixes. please see original code I wrote 
https://github.com/microbiomedata/nmdc-schema/blob/55f087ad90982d647b82f42efd2662f8368740ba/nmdc_schema/napa_study_biosample_omics_migration.py#L66